### PR TITLE
Migrate `@woocommerce/data` import & export stores to TS

### PR DESCRIPTION
**Pull request by chihsuan**
### All Submissions:

-   [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Part of #33105 

This PR migrates `import` and `export` stores to TS

### How to test the changes in this Pull Request:

1. Should run `pnpm nx build @woocommerce/data` without errors
3. Should run `pnpm nx test @woocommerce/data` without errors

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm nx changelog <project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/data/src/import/types.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+export type ImportStatus =
+	| isImporting
+	| ( isImporting & SchedulerImportStatus );
+
+export type ImportStatusQuery = number;
+
+export type ImportTotals = {
+	[ schedulerName in SchedulerName ]: number;
```
Comment (chihsuan) [inline packages/js/data/src/import/types.ts:23]: ref: https://github.com/woocommerce/woocommerce/blob/d0f994ffe63817400f8ad0fb0db3863a2fa12bc8/plugins/woocommerce/src/Admin/ReportsSync.php#L148

File: packages/js/data/src/import/types.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+type SchedulerName = 'orders' | 'customers';
```
Comment (chihsuan) [inline packages/js/data/src/import/types.ts:1]: ref: https://github.com/woocommerce/woocommerce/blob/d0f994ffe63817400f8ad0fb0db3863a2fa12bc8/plugins/woocommerce/src/Admin/ReportsSync.php#L42-L43

File: packages/js/data/src/import/types.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+};
+
+type SchedulerImportStatus = {
+	[ schedulerName in SchedulerName ]: {
+		imported: number;
+		totals: number;
+	};
+} & {
+	imported_from?: string | number;
+};
```
Comment (chihsuan) [inline packages/js/data/src/import/types.ts:14]: ref:
https://github.com/woocommerce/woocommerce/blob/d0f994ffe63817400f8ad0fb0db3863a2fa12bc8/plugins/woocommerce/src/Admin/ReportsSync.php#L104-L122

File: packages/js/data/src/import/types.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+export type ImportStatus =
+	| isImporting
+	| ( isImporting & SchedulerImportStatus );
```
Comment (chihsuan) [inline packages/js/data/src/import/types.ts:18]: ref: https://github.com/woocommerce/woocommerce/blob/d0f994ffe63817400f8ad0fb0db3863a2fa12bc8/plugins/woocommerce/src/Admin/ReportsSync.php#L129-L134

File: packages/js/data/src/import/types.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+export type ImportStatusQuery = number;
```
Comment (chihsuan) [inline packages/js/data/src/import/types.ts:20]: https://github.com/woocommerce/woocommerce/blob/62aafcdd3f34e3f887b40cd103dcbb6e2a5da3f7/plugins/woocommerce-admin/client/analytics/settings/historical-data/layout.js#L132

ImportStatusQuery -> lastImportStartTimestamp: number

File: packages/js/data/src/import/types.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+export type ImportTotals = {
+	[ schedulerName in SchedulerName ]: number;
+};
+
+export type ImportTotalsQuery = {
+	skip_existing: boolean;
+	days: number;
+};
```
Comment (chihsuan) [inline packages/js/data/src/import/types.ts:29]: ref:
https://github.com/woocommerce/woocommerce/blob/62aafcdd3f34e3f887b40cd103dcbb6e2a5da3f7/plugins/woocommerce/src/Admin/API/Reports/Import/Controller.php#L185-L202

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33155/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Add changelog` 8c7533a7e8b38f2054b7fbe7cc4688c5a17adceb
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

**Review by ilyasfoo — DISMISSED** [review]
Tests well, LGTM! 🚢

**Pull request merged**